### PR TITLE
OSDOCS-8044 - adding openshift-docs build to CI

### PR DESCRIPTION
**Pull request by aireilly**
Adding openshift-docs build to CI. 

This initial PR targets the `prow-test` branch in /openshift-docs repo. Once user acceptance testing is complete we will enable for main and other production branches. 

https://issues.redhat.com/browse/OSDOCS-8044

### CODE REVIEW

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-ocp-docs

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-ocp-docs

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-ocp-docs

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciibinder-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciidoctor-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciibinder-build
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciidoctor-build
/pj-rehearse pull-ci-openshift-openshift-docs-main-portal-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciibinder-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciidoctor-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-portal-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciibinder-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciibinder-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciibinder-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-asciidoctor-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-portal-build

**Comment by aireilly** [comment]
/pj-rehearse /pj-rehearse pull-ci-openshift-openshift-docs-main-build

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-deploy-preview

File: ci-operator/step-registry/openshift-docs/openshift-docs-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+ref:
+  as: openshift-docs
+  from: base
```
Comment (aireilly) [inline ci-operator/step-registry/openshift-docs/openshift-docs-ref.yaml:3]:
```suggestion
  from: openshift-docs-builder
```

File: ci-operator/step-registry/openshift-docs/openshift-docs-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  commands: openshift-docs-commands.sh
+  credentials:
+  - mount_path: /var/run/openshift-docs-netlify-secret
+    name: netlify-secret
+    namespace: test-credentials
+  resources:
+    requests:
+      cpu: 100m
+      memory: 100Mi
```
Comment (aireilly) [inline ci-operator/step-registry/openshift-docs/openshift-docs-ref.yaml:12]: Need to set
`env:` for the netlify secret.

Comment (aireilly) [reply ci-operator/step-registry/openshift-docs/openshift-docs-ref.yaml:12]: Or not use the ref at all. https://docs.ci.openshift.org/docs/how-tos/adding-a-new-secret-to-ci/#use-a-secret-in-non-step-jobs

**Comment by aireilly** [comment]
pj-rehearse pull-ci-openshift-openshift-docs-main-build

**Comment by aireilly** [comment]
pj-rehearse pull-ci-openshift-openshift-docs-main-deploy-preview

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build pj-rehearse pull-ci-openshift-openshift-docs-main-deploy-preview

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

File: ci-operator/step-registry/openshift-docs/asciidoc/openshift-docs-asciidoc-ref.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+ref:
+  as: openshift-docs-asciidoc
+  commands: openshift-docs-asciidoc-commands.sh
+  resources:
+    requests:
+      cpu: 100m
+      memory: 100Mi
+  from: src
```
Comment (aireilly) [inline ci-operator/step-registry/openshift-docs/asciidoc/openshift-docs-asciidoc-ref.yaml:8]: container:
  from: pipeline:src

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by danilo-gemoli** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-prow-test-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-prow-test-images

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-main-images pull-ci-openshift-openshift-docs-prow-test-images

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-prow-test-images

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-prow-test-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-prow-test-topic-maps

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-prow-test-topic-maps

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-prow-test-build-enterprise

**Comment by aireilly** [comment]
/pj-rehearse pull-ci-openshift-openshift-docs-prow-test-build-enterprise

**Comment by aireilly** [comment]
squashed commits

**Comment by kalexand-rh** [comment]
LGTM

**Pull request merged**